src/background/mod.rs

```diff
@@ -41,18 +70,176 @@ pub struct PlanetComponent {
     pub rotation_speed: f32,
 }
 
+#[derive(Component)]
+pub struct StarLightComponent;
+
 /// Rotate planets about their z axis
-pub fn rotate_planet_system(mut query: Query<(&mut Transform, &PlanetComponent)>) {
+pub fn rotate_planet_system(mut query: Query<(&mut Transform, &PlanetComponent)>, time: Res<Time>) {
     for (mut transform, planet) in query.iter_mut() {
-        transform.rotation *= Quat::from_rotation_y(planet.rotation_speed);
+        transform.rotation *= Quat::from_rotation_y(planet.rotation_speed * time.delta_seconds());
+    }
+}
+
+/// Execute the exploding star effect if the game is lost through defense being destroyed
+pub fn on_defeat_star_explode_system(
+    mut run_end_event_reader: EventReader<RunEndEvent>,
+    mut point_light_query: Query<&mut PointLight, With<StarLightComponent>>,
+    mut star_explode_res: ResMut<StarExplodeResource>,
+    time: Res<Time>,
+    backgrounds_res: Res<BackgroundsResource>,
+) {
+    for event in run_end_event_reader.iter() {
+        if let RunOutcomeType::Defeat(RunDefeatType::DefenseDestroyed) = event.outcome {
+            star_explode_res.started = true;
+        }
+    }
+
+    if star_explode_res.started {
+        for mut point_light in point_light_query.iter_mut() {
+            point_light.intensity *= backgrounds_res.star_explode_intensity * time.delta_seconds();
+        }
     }
 }
 
 /// Create background from resource
 pub fn create_background_system(
     mut commands: Commands,
     asset_server: Res<AssetServer>,
-    backgrounds: Res<BackgroundsResource>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut star_explode_res: ResMut<StarExplodeResource>,
+    backgrounds_res: Res<BackgroundsResource>,
 ) {
-    backgrounds.backgrounds["solar_system"].spawn(&mut commands, &asset_server);
+    // reset the star explode reource
+    *star_explode_res = StarExplodeResource::default();
+
+    let mut rng = rand::thread_rng();
+
+    // positions
+    let background_transform = Transform::from_translation(backgrounds_res.background_transation);
+    let star_transform = Transform::from_xyz(
+        rng.gen_range(backgrounds_res.star_position_x_range.clone()),
+        0.0,
+        rng.gen_range(backgrounds_res.star_position_z_range.clone()),
+    );
+    let planet_transform = Transform::from_translation(backgrounds_res.planet_translation);
+
+    // randomly generate attributes
+    let random_planet_file = fs::read_dir("./assets/models/planets")
+        .unwrap()
+        .choose(&mut rng)
+        .unwrap()
+        .unwrap()
+        .path();
+    let random_planet_filename = random_planet_file.file_name().unwrap().to_str().unwrap();
+
+    let random_background_file = fs::read_dir("./assets/texture/backgrounds")
+        .unwrap()
+        .choose(&mut rng)
+        .unwrap()
+        .unwrap()
```

Comment:
Let's get of 2 of these unwraps by having a simple static gradient or single color background if none of the assets can be found. Log and use the fallback rather than panic.